kernel/thread/worker.c

```diff
@@ -29,7 +29,7 @@ static void *thd_worker_thread(void *d) {
     for (;;) {
         flags = irq_disable();
 
-        if (!worker->pending)
+        if ((!worker->pending) && (!worker->quit))
```

Comment:
Documenting the case, since I won't remember later:
1A. The worker was woken and has started it's routine, setting worker->pending to false
2B. A thread/interrupt calls `thd_worker_destroy(worker), setting worker->quit to true (and genwait_wake_one, which fails as the worker isn't waiting)
3A. Worker A finishes its routine, looping, and reaches if !pending so it genwaits and deadlocks.